src/v3/aes.rs

```diff
@@ -0,0 +1,89 @@
+use super::{verify_signature, IV_LEN};
+use crate::{
+    aes::{decrypt_attached_document_core, EncryptionKey, PlaintextDocument},
+    EncryptedPayload, Error, MAGIC,
+};
+use protobuf::Message;
+
+// [3, b"IRON", <2 bytes of header len>]
+const DETACHED_HEADER_LEN: usize = 7;
+
+const V3: u8 = 3u8;
+
+/// Decrypt a V3 detached document and verifies its signature.
+pub fn decrypt_detached_document(
+    key: &EncryptionKey,
+    payload: EncryptedPayload,
+) -> Result<PlaintextDocument, Error> {
+    let payload_len = payload.0.len();
+    if payload_len < DETACHED_HEADER_LEN + IV_LEN {
+        Err(Error::EdocTooShort(payload_len))?
+    };
+    let (magic_header, header_and_len_and_cipher) = payload.0.split_at(5);
+    if magic_header != [&[V3], &MAGIC[..]].concat() {
+        Err(Error::NoIronCoreMagic)?
+    };
+    let (header_len_bytes, header_and_cipher) = header_and_len_and_cipher.split_at(2);
```

Comment:
Constant here as well. DETACHED_HEADER_LEN will be the constant you created for this and the one created above added together.